packages/js-config-webpack-plugin/src/JsConfigWebpackPlugin.js

```diff
@@ -33,13 +33,32 @@ class JsConfigWebpackPlugin {
 	 * @returns {string}
 	 */
 	resolveBabelConfigFilePath(contextPath, environmentName) {
+		/**
+		 * Return the value, unless its an interator, in which case get the last value.
+		 * Works around differences in return values of @babel/core functions between versions.
+		 * @template T
+		 * @param {Generator<T> | T} valueOrIterator
+		 * @returns {T}
+		 */
+		// prettier-ignore
+		function getResult(valueOrIterator) {
+			if (valueOrIterator && typeof /** @type {Generator<T>} */ (valueOrIterator).next === 'function') {
+				let result = /** @type {Generator<T>} */ (valueOrIterator).next();
+				while (!result.done) {
+					result = /** @type {Generator<T>} */ (valueOrIterator).next();
+				}
+				return result.value;
+			}
+			return /** @type {T} */ (valueOrIterator);
+		}
+
 		// From https://github.com/babel/babel/blob/52a569056c6008c453bf26219461655c7d0322c4/packages/babel-core/src/config/files/package.js#L15
```

Comment:
Typescript has a helper to allow you writing custom type guards:

```ts
const extractPackageData = <T extends {}>(valueOrIterator: Generator<T,T> | T): T => {
    if (!isGenerator(valueOrIterator)) {
        return valueOrIterator;
    }
    let result = valueOrIterator.next();
    while (!result.done) {
        result = valueOrIterator.next();
    }
    return result.value;
}

const isGenerator = (val: any): val is Generator => {
    // maybe we can find a better check here?
    return val && typeof val === 'object' && typeof val.next === 'function';
}
```

[Playground Link](https://www.typescriptlang.org/play?#code/MYewdgzgLgBApgDygJwIbCgBXQa1QczgBFUpUYBeGAHgBV4k4wATCGAbwF8A+ACgDdUAGwCucAPLIAklDhooIZAC4YAcSZzSiugBpa3GAB8YtAJQr6FA+wBQMezACWAMxi8AhI4jqwmhcgFhMUkZP0VTUw47BxjkOCgRZDAYQVEJaVl5RQBuaPtOPJgheJg4iBEhWCpU4Iyw5AA6XyReU1yYgHcAC0dit3cyiqgG5nA4SNsY2LhyysoUoPTQrMbmqFb2hwLphKTSmaGGmrhcgptQSFgvH3r5wKEVVDAAT3MFoSc2G5XKa0K43bJVIwABkIJgUGeAAc4CBXMCKIiYAByEAAIwAVnAMMjQeDITC4e8mogqkjkc4RGAMI5wMjTkA)

Reply:
`JsConfigWebpackPlugin.js` isn't TypeScript, but `tsc` is run as part of the linting step it looks like it will enforce whatever it finds in the JSDocs. Should `JsConfigWebpackPlugin.js` be turned into TypeScript? Or do you know if there a JSDoc alternative I could use?

Reply:
But I can definitely add your improvement to the check :+1: